pkg/csi/cinder/controllerserver.go

```diff
@@ -420,10 +420,17 @@ func (cs *controllerServer) CreateSnapshot(ctx context.Context, req *csi.CreateS
 		}
 
 		if len(backups) == 1 {
-			backup = &backups[0]
+			// since backup.VolumeID is not part of ListBackups response
+			// we need fetch single backup to get the full object.
+			backup, err = cs.Cloud.GetBackupByID(backups[0].ID)
```

Comment:
ListBackups filters by Name and the response is https://docs.openstack.org/api-ref/block-storage/v3/index.html#list-backups-for-project

There is another list backups api that will response all of the details https://docs.openstack.org/api-ref/block-storage/v3/index.html#list-backups-with-detail However, that will not support filter by Name. So I think fetching this single backup by id is enough for now.